File: shesha-core/src/Shesha.Application/FormTemplates/DetailsView/DetailsViewExtensionJson.cs
using System.Collections.Generic;

namespace Shesha.FormTemplates.DetailsView
{
    public class DetailsViewExtensionJson
    {
        public string ModelType { get; set; }
        public bool ShowKeyInformationBar { get; set; }
        public List<string> KeyInformationBarProperties { get; set; } = new List<string>();
        public bool AddChildTables { get; set; }
        public List<string> ChildTablesList { get; set; } = new List<string>();
    }
}

File: shesha-reactjs/src/providers/sheshaApplication/publicApi/forms/models/DetailsViewExtensionJson.ts
import { IEntityTypeIdentifier } from "../../entities/models";

export interface DetailsViewExtensionJson {
  modelType: string | IEntityTypeIdentifier;
  showKeyInformationBar: boolean;
  keyInformationBarProperties: string[] | undefined;
  addChildTables: boolean;
  childTablesList: string[];
}
